modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -803,6 +797,14 @@ public CompletableFuture<Ignite> start(Path configPath) {
 
                         return metaStorageMgr.recoveryFinishedFuture();
                     }, startupExecutor)
+                    .thenComposeAsync(revision -> {
```

Comment:
What if the entire lambda becomes `initializeClusterConfiguration`? Less lines, easier to read...